### ISSUE TITLE
feat(learning_rate): Add LinearWarmupLR — linear ramp then constant

### ISSUE THREAD

**Issue by sanskarpan** [sanskarpan/Optimization#33]
## Description
Linear warm-up from `start_lr` to `initial_lr` over `warmup_steps`, then constant.

```
lr(t) = start_lr + (initial_lr - start_lr) * t / warmup_steps   if t < warmup_steps
lr(t) = initial_lr                                                 if t >= warmup_steps
```

## Acceptance Criteria
- [ ] Starts at `start_lr` (default 0.0) at step 0
- [ ] Reaches `initial_lr` exactly at step `warmup_steps`
- [ ] Constant after `warmup_steps`
- [ ] `warmup_steps <= 0` raises `ValueError`
